components/suggestion/modal_suggestion_list.jsx

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import SuggestionList from 'components/suggestion/suggestion_list.jsx';
+import {getClosestParent} from 'utils/utils.jsx';
+
+export default class ModalSuggestionList extends React.PureComponent {
+    static propTypes = {
+        location: PropTypes.string.isRequired,
+        open: PropTypes.bool.isRequired,
+        cleared: PropTypes.bool.isRequired,
+        calculateInputRect: PropTypes.func.isRequired,
+        onLoseVisibility: PropTypes.func.isRequired,
+    }
+
+    latestHeight = 0;
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            scroll: 0,
+            modalBounds: {top: 0, bottom: 0},
+            inputBounds: {top: 0, bottom: 0, width: 0},
+            location: props.location,
+        };
+
+        this.container = React.createRef();
+        this.suggestionList = React.createRef();
+    }
+
+    onModalScroll = (e) => {
+        if (this.state.scroll !== e.target.scrollTop &&
+            this.latestHeight !== 0) {
+            this.setState({scroll: e.target.scrollTop});
+        }
+    }
+
+    componentDidMount() {
+        if (this.container.current) {
+            const modalBodyContainer = getClosestParent(this.container.current, '.modal-body');
+            modalBodyContainer.addEventListener('scroll', this.onModalScroll);
```

Comment:
this seems _very_ brittle. Would you be able to figure out where this component is if you received x,y,width of the input component? if so i would recommend using that this scroll change events.

Reply:
I can figure out the position of the component if I have x, y, and width, yes. The problem is that whenever the modal scrolls, those values would change. if we could pass x, y and width as props instead of the calculateInputRect function, there would be no need for checking this scroll.

Also, this class name is not something we can control. It comes from react-bootstrap's Modal. Which makes it a bit more robust (we cannot change that class name unless we use something different as a modal) and brittle at the same time (any change on react-bootstrap's Modal that change that will break the application).

Nevertheless, I think it would be very difficult to pass these values through props. Since it is more related to the previous comment, I will explain there.